src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -266,6 +267,18 @@ export class RecommendationHandler {
                 requestId = error.requestId || ''
                 reason = `CodeWhisperer Invocation Exception: ${error?.code ?? error?.name ?? 'unknown'}`
                 await this.onThrottlingException(error, triggerType)
+
+                if (error?.code === 'AccessDeniedException') {
+                    getLogger().error('CodeWhisperer AccessDeniedException : %s', (error as Error).message)
```

Comment:
Are there any cases when this AccessDenied is not coming from IAM? Should we also check the errorMessage here?

Reply:
@andrewyuq I would not agree on parsing errorMessages as they can change in service side breaking dependencies. Any other way to detect IAM error here?

Reply:
Unfortunately this is the currently the only way we check it's due to IAM. All plugin receives is an AccessDeniedException with the IAM message.

Reply:
yes, the other cases in the codebase that check AccessDeniedException check other parts of the message: 

- https://github.com/aws/aws-toolkit-vscode/blob/13ebf3aee88b2357a9a59b27d0308417b3f105c0/src/codecatalyst/auth.ts#L215
- https://github.com/aws/aws-toolkit-vscode/blob/13ebf3aee88b2357a9a59b27d0308417b3f105c0/src/codewhisperer/service/recommendationHandler.ts#L305
- https://github.com/aws/aws-toolkit-vscode/blob/13ebf3aee88b2357a9a59b27d0308417b3f105c0/src/shared/clients/codecatalystClient.ts#L235

...so checking `AccessDeniedException` without also checking for message contents, could be a potential bug.

it's also probably time to extract this common logic into a shared function in [errors.ts](https://github.com/aws/aws-toolkit-vscode/blob/13ebf3aee88b2357a9a59b27d0308417b3f105c0/src/shared/errors.ts) , something like `errors.ts:isAccessDeniedError()`. It could live next to [this](https://github.com/aws/aws-toolkit-vscode/blob/13ebf3aee88b2357a9a59b27d0308417b3f105c0/src/shared/errors.ts#L373). This function may not do much but it serves as a placeholder for all of the places that are using this bit of esoteric knowledge.